packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/relay/relay.ts

```diff
@@ -25,8 +25,22 @@ const connections = config.solanaEndpoints.map(
   (endpoint) => new Connection(endpoint)
 )
 
-const delay = async (ms: number) => {
-  return await new Promise((resolve) => setTimeout(() => resolve, ms))
+const delay = async (ms: number, options?: { signal: AbortSignal }) => {
+  const signal = options?.signal
+  return new Promise<void>((resolve, reject) => {
+    if (signal?.aborted) {
+      reject()
+    }
+    const listener = () => {
+      clearTimeout(timer)
+      reject()
+    }
+    const timer = setTimeout(() => {
+      signal?.removeEventListener('abort', listener)
+      resolve()
+    }, ms)
+    signal?.addEventListener('abort', listener)
+  })
```

Comment:
Neat

Reply:
Also not sure what the layout is in this package, but seems like utils should go outside this module?

Reply:
That's fair. I think I like it co-located for now, but if this module grows a bit more or other modules want to use it I'll bring it out